protocol/contracts/v2/strategies/RamosStrategy.sol

```diff
@@ -0,0 +1,137 @@
+import { AMO__IRamos } from "contracts/amo/interfaces/AMO__IRamos.sol";
+import { AMO__IAuraStaking } from "contracts/amo/interfaces/AMO__IAuraStaking.sol";
+import { AMO__IPoolHelper } from "contracts/amo/interfaces/AMO__IPoolHelper.sol";
+
+contract RamosStrategy  is AbstractStrategy {
+    using SafeERC20 for IERC20;
+    string public constant VERSION = "1.0.0";
+
+    /**
+     * @notice The RAMOS contract used to manage the TPI
+     */
+    AMO__IRamos public ramos;
+
+    /**
+     * @notice The list of assets which are reporting on for equity performance updates
+     * Total balances = the RAMOS strategy balance of the asset + any delta from `assetBalanceDeltas`
+     */
+    address[] public assets;
+
+    event AssetsSet(address[] _assets);
+    event Borrow(uint256 amount);
+    event Repay(uint256 amount);
+
+    constructor(
+        address _initialRescuer,
+        address _initialExecutor,
+        string memory _strategyName,
+        address _treasuryReservesVault,
+        address _ramos
+    ) AbstractStrategy(_initialRescuer, _initialExecutor, _strategyName, _treasuryReservesVault) {
+        ramos = AMO__IRamos(_ramos);
+    }
+
+    /**
+     * The version of this particular strategy
+     */
+    function strategyVersion() external override pure returns (string memory) {
+        return VERSION;
+    }
+
+    /**
+     * @notice The assets on which to report balances need to be set by the Strategy Executors
+     * @dev Use the zero address (0x000) to represent native ETH
+     */
+    function setAssets(address[] calldata _assets) external onlyElevatedAccess {
+        assets = _assets;
+        emit AssetsSet(_assets);
+    }
+
+    /**
+     * @notice The assets on which to report balances need to be set by the Strategy Executors
+     * @dev The zero address (0x000) represents native ETH
+     */
+    function getAssets() external view returns (address[] memory) {
+        return assets;
+    }
+
+    /**
+     * @notice The latest checkpoint of each asset balance this stratgy holds, and the current debt.
+     * This will be used to report equity performance: `sum(asset value in STABLE) - debt`
+     * The conversion of each asset price into the stable token (eg DAI) will be done off-chain
+     *
+     * @dev The asset value may be stale at any point in time, depending onthe strategy. 
+     * It may optionally implement `checkpointAssetBalances()` in order to update those balances.
+     */
+    function latestAssetBalances() public override view returns (AssetBalance[] memory assetBalances, uint256 debt) {
+        // RAMOS strategy assets = RAMOS's DAI balance + claimed AURA & BPT rewards =
+        // (bpt.balanceOf(RAMOS) / bpt.totalSupply() * Total_DAI_Balance in the LP) + 
+        // claimed AURA & BAL rewards
+
+        // get the RAMOS's DAI balance
+        AMO__IAuraStaking amoStaking = AMO__IAuraStaking(ramos.amoStaking());
```

Comment:
We may want to consider storing these locally as immutables to save paying for gas each time they're needed. They'll be needed for on-chain tasks too later

Downside is that if the contracts change we'll need to change here too. But I'm coming around the opinion that we could make these immutable, and if they need updating then it would mean we just push a new ramos strategy out.

Thoughts?

Reply:
Since RAMOS contract has the `setAmoStaking()` function, I think it's better to avoid the contract address mismatched by pulling `amoStaking` contract address from RAMOS contract.

Reply:
Sure - i'm ok either way. As an intermediate we could consider adding a single function to RAMOS which returns all addresses in one call (if needed)

Can think about it after the other functions are implemented